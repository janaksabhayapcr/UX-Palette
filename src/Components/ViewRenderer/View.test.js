import View from './View';
import React from 'react';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';
import { configure, shallow, wrapper, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { toast } from 'react-toastify';
import { MemoryRouter as Router, MemoryRouter } from 'react-router-dom';

configure({ adapter: new Adapter() });

// jest mock functions (mocks this.props.func)
const extractDataFromXML = jest.fn(() => {
  return [
    {
      type: [{ _: '' }],
      renderertype: [],
      viewtitle: [],
      catalog: [
        {
          resolvedtemplate: ['test'],
        },
      ],
    },
  ];
});
const deleteView = jest.fn();
const toggleSettings = jest.fn();
const onClick = jest.fn();
const onChange = jest.fn();
// defining this.props
const baseProps = {
  sideToRender: {},
  renderLocation: {},
  viewXML: {},
  extractDataFromXML,
  universalView: {},
  context: [
    {
      Parameter: {},
      Value: {},
    },
  ],
  user: {},
  chartPrototype: {},
  chartConfig: {},
  initialLoad: {},
  viewChartScript: {},
  logSuccessfulChart: {},
  liquidFunctions: {},
  layoutViewId: {},
  getCleanJSON: {},
  ViewScript: {},
  liquidTemplates: {},
  preRenderedHTML: {},
  layoutViewId: {},
  services: {
    Dashboard: {
      URL: {},
      subscription_key: {},
    },
  },
  deleteView,
  toggleSettings,
  userRoles: {},
  allowEdit: true,
  location: {
    pathname: [],
  },
  viewId: {},
  viewType: {},
  layoutViewId: {},

  defaultData: {
    defaultView: [],
    size: [],
    url: [],
    description: [[]],
    viewtitle: [[]],
    categories: [],
    urlcaption: {},
  },
  localPermissions: [
    {
      globals: {
        UI_LAYOUT_VIEW_EDIT: [],
        UI_LAYOUT_VIEW_INSTANCE: [],
        UI_LAYOUT_EDIT: [],
      },
    },
  ],
  drilldownRender: true,
  onClick,
  onChange,
};

describe('View Renderer : View Test', () => {
  let tree;
  let wrapper;
  beforeEach(
    () =>
      (wrapper = mount(
        <MemoryRouter keyLength={0}>
          <View {...baseProps} />
        </MemoryRouter>
      ))
  );

  it(' Should render with all of the props', () => {
    tree = renderer.create(
      <MemoryRouter>
        <View {...baseProps} />
      </MemoryRouter>
    );
    let treeJson = tree.toJSON();
    expect(treeJson).toMatchSnapshot();
    tree.unmount();
  });
  //check
  it('should call toggleViewType method on button click', () => {
    baseProps.onClick.mockClear();
    wrapper.setProps({});
    wrapper.find('View').setState({
      defaultView: 'chart',
      printDivId: 'print',
    });
    wrapper.update();
    wrapper.find('View').find('#view-rotate-with-link').at(1).simulate('click');
  });
  //check
  it('should call populateViewDropdown method on button click', () => {
    baseProps.onClick.mockClear();
    wrapper.setProps({});
    wrapper.find('View').setState({});
    wrapper.update();
    wrapper.find('View').find('#view-title-img').at(0).simulate('click');
  });

  it('Test deleteView function: Notification: View Deleted', () => {
    let options = {
      type: toast.TYPE.SUCCESS,
      position: toast.POSITION.TOP_LEFT,
    };
    toast('View Deleted', options);
    wrapper.setState({
      layoutViewId: '',
    });

    wrapper.update();
    expect(wrapper.find('View').instance().deleteView()).toEqual();
    expect(wrapper.find('View').instance().deleteView()).toBeUndefined();
  });

  it('Test deleteView function: Notification: Failed to Delete', () => {
    let options = {
      type: toast.TYPE.ERROR,
      position: toast.POSITION.TOP_LEFT,
    };
    toast('Failed to Delete', options);

    wrapper.update();
    expect(wrapper.find('View').instance().deleteView()).toEqual();
    expect(wrapper.find('View').instance().deleteView()).toBeUndefined();
  });
});
