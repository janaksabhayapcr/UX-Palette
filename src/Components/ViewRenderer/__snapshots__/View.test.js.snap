// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Renderer : View Test  Should render with all of the props 1`] = `
<div
  className="centered-view-container"
>
  <div
    className="uw-4x4"
    id="view-container"
  >
    <div
      id="title-container"
    >
      <div
        className="view-titlediv"
        title="View-Layout ID: [object Object] / View ID: [object Object]"
      >
        <h2
          id="view-title"
          title="View Instance of Title"
        >
          Title
          <div />
        </h2>
      </div>
      <div
        className="view-externalLink"
        title={Object {}}
      >
        <div
          onClick={[Function]}
          title={Object {}}
        >
          <span
            aria-hidden={true}
            className="fa fa-view-externallink-img fa-2x fa fa-external-link view-externallink-img"
            style={
              Object {
                "color": "lightgray",
                "textShadow": "0 1px 0 rgba(0, 0, 0, 0.1)",
              }
            }
          />
        </div>
      </div>
      <span
        aria-hidden={true}
        className="fa fa-view fa-lg fa-line-chart"
        id="view-rotate-with-link"
        onClick={[Function]}
        title="View Chart"
      />
      <div
        className="view-menu"
      >
        <ul
          aria-activedescendant=""
          className="rc-menu rc-menu-vertical  rc-menu-root"
          id="view-title-img"
          onKeyDown={[Function]}
          role="menu"
          style={Object {}}
          tabIndex="0"
        >
          <li
            className="view-sub-menu rc-menu-submenu-vertical rc-menu-submenu"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={undefined}
          >
            <div
              aria-expanded={false}
              aria-haspopup="true"
              aria-owns="item_0$Menu"
              className="rc-menu-submenu-title"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
              title={undefined}
            >
              <div
                className="view-sub-menu-img"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-menu-bar-gears fa-bars view-menu-img"
                />
              </div>
              <i
                className="rc-menu-submenu-arrow"
              />
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="__react_component_tooltip place-top type-dark "
      data-id="tooltip"
    >
      
    </div>
  </div>
</div>
`;
